### PR TITLE
gui: add error message to launcher

### PR DESCRIPTION
**Pull request by edouardparis**
Gui will do a check that the selected network datadir is correct and will display an error message in case.

### CODE REVIEW

**Comment by edouardparis** [comment]
2023-02-07T16:31:03,240527706+01:00

**Comment by edouardparis** [comment]
2023-02-07T16:40:17,322545080+01:00

File: gui/src/installer/view.rs [hunk 1]
```diff
@@ -376,7 +376,7 @@ pub fn import_descriptor<'a>(
             form::Form::new("Descriptor", imported_descriptor, |msg| {
                 Message::DefineDescriptor(message::DefineDescriptor::ImportDescriptor(msg))
             })
-            .warning("Please enter correct descriptor")
+            .warning("Please enter compatible descriptor (Liana >0.2 does not support descriptor with keys without origins)")
```
Comment (darosior) [inline gui/src/installer/view.rs:379]:
```suggestion
            .warning("Incompatible descriptor. Note that starting from v0.2 Liana requires extended keys in a descriptor to have an origin.")
```

**Review by darosior — COMMENTED** [review]
Just a wording nit, i still need to have a look to the datdir deletion logic..

File: gui/src/launcher.rs [hunk 2]
```diff
@@ -42,71 +48,163 @@ impl Launcher {
...
+            } else {
+                format!(
+                    "Failed to read daemon configuration file in the directory: {}",
+                    cfg.daemon_config_path.to_string_lossy()
+                )
+            }
+        }
+        ConfigError::UnexpectedDescriptor(_) => {
+            "There is an issue with the configuration for this network. You most likely use a descriptor containing one or more public key(s) without origin. Liana v0.2 and later only support public keys with origins. Please migrate your funds using Liana v0.1.".to_string()
+        }
```
Comment (darosior) [inline gui/src/launcher.rs:200]: This should be removed from the daemon in the future. It is never returned at the moment. (Do not modify this code though)

**Review by darosior — APPROVED** [review]
Ack the wording. `check_network_datadir` looks correct although i don't know how the state transition works in the GUI.

Sorry to have you do this ugly check, i should have thought about origins before releasing v0.1. :(

**Comment by darosior** [comment]
Did you test this?

**Comment by darosior** [comment]
ACK 3f3e6fab03cfad7219d0860dd096d84a31667d7f -- quickly tested. See below.

image

image

**Pull request merged**